grype/db/v5/namespace/index.go

```diff
@@ -12,6 +13,8 @@ import (
 	syftPkg "github.com/anchore/syft/syft/pkg"
 )
 
+var alpineVersionRegularExpression = regexp.MustCompile(`^(\d+)\.(\d+)(?:\.(\d+))?(?:-r(\d+))?$`)
```

Comment:
I think this regex should be changed to `^(\d+)\.(\d+)\.(\d+)$`. As written the regex will match version strings beyond the `x.y.z` rule, such as `3.16` and `1.2.3-r4`.

Reply:
^ Ahh yea - I thought that `x.y.z` would also include `x.y` as a derivative. I'll update this to be more strict and add a case for the r4